File: Frontend/Managment-FE/src/app/pages/models/product.ts
export interface Product {
    id: string; 
    name: string;
    description: string;
    price: number; 
    stockQuantity: number;
    categoryId: string;
    categoyName: string;
    imageUrl?: string | null; 
    createdDate: string; 
    updatedDate: string; 
    status: ProductStatus; 
    isEditable?: boolean; 

  }
  export interface PaginatedProduct {
    totalCount: number; 
    pageNumber: number; 
    pageSize: number; 
    data: Product[]; 
  }
  export enum ProductStatus {
    Active = 0,
    Inactive = 1,
    Discontinued = 2
  }
  

File: Backend/Management.DL/Models/Product.cs
﻿using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Management.DL.Models
{
    public class Product
    {
        public Guid Id { get; set; }
        public string Name { get; set; } = string.Empty;
        public string Description { get; set; } = string.Empty;
        public decimal Price { get; set; }
        public Guid CategoryId { get; set; }
        public Category? Category { get; set; }
        public ProductStatus Status { get; set; }
        public int StockQuantity { get; set; }
        public string? ImageUrl { get; set; }
        public DateTime CreatedDate { get; set; }
        public DateTime UpdatedDate { get; set; }
    }

}
